src/Symfony/Contracts/Service/ServiceLocatorTrait.php

```diff
@@ -60,12 +62,22 @@ public function get($id)
 
         $this->loading[$id] = $id;
         try {
-            return $this->factories[$id]($this);
+            return $this->factories[$key]($this);
         } finally {
             unset($this->loading[$id]);
         }
     }
 
+    private function getWithoutKey($id) {
+        foreach ($this->factories as $key => $value) {
+            if ($id === get_class($value())) {
```

Comment:
instantiating all the objects to get the class is a no-go from a performance PoV.